File: DWES-Hugo-Moruno/Spring/viajes-clase/src/main/java/com/daw2/viajesclase/models/entity/Cliente.java
package com.daw2.viajesclase.models.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name="clientes")
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(unique = true, length = 12)
    private String nif;

    @Column(nullable = false, length = 20)
    private String nombre;

    @Column(nullable = false, length = 20)
    private String apellido1;

    @Column(nullable = true, length = 20)
    private String apellido2;

    @Column(nullable = true)
    private Date fecha_nacimiento;

    @Column(nullable = true, length = 200)
    private String foto;

    @Lob
    private String observaciones;

    @Column()
    private Boolean active = true;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
}

File: DWES-Hugo-Moruno/Laravel/aprende03/proyectos/proyectViajes/app/Models/Cliente.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Cliente extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'clientes';

    protected $fillable = [
        'nif',
        'nombre',
        'apellido1',
        'apellido2',
        'fecha_nacimiento',
        'foto',
        'observaciones',
        'active',
    ];
}
